scripts/bump.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+set -eu -o pipefail
+cd "$(dirname "$0")"/..
+
+log () { echo "[$(date +%Y-%m-%dT%H:%M:%S%Z)]" "$@"; }
+
+: ${BRANCH_TO_BUILD:=main}
+: ${PR_REVIEWERS:=mattermost/core-build-engineers}
+: ${BUMP_BUILD_NUMBER:=}
+: ${BUMP_VERSION_NUMBER:=}  # If enabled, you must populate the VERSION variable as well
+: ${CREATE_PR:=}            # Enable CREATE_PR to push the commit to origin, and create a corresponding PR
+: ${PR_EXTRA_MESSAGE:=}     # Optional message to add in the PR description
+: ${GIT_LOCAL_BRANCH:=chore-bump-${BRANCH_TO_BUILD}-$(date +%s)}
+
+log "Checking that the configuration is sane"
+if [ -n "$BUMP_VERSION_NUMBER" ]; then
+  : ${VERSION:?Setting this variable is required when BUMP_VERSION_NUMBER is set.}
+fi
+
+log "Asserting that the workdir is clean"
+if ! git diff --quiet; then
+  log "Error: workdir is not clean. Aborting" >&2
+  exit 1
+fi
+
+log "Saving the currently checked out branch"
+CURRENT_BRANCH=$(git branch --show-current)
+trap "git checkout $CURRENT_BRANCH" EXIT
+
+log "Creating branch '${GIT_LOCAL_BRANCH}' based on branch '$BRANCH_TO_BUILD'"
+git checkout $BRANCH_TO_BUILD
+git pull
+git checkout -b $GIT_LOCAL_BRANCH
+
+if [ -n "$BUMP_BUILD_NUMBER" ]; then
+  log "Selecting the next largest build number..."
+  LATEST_BUILD_NUMBER=$(./scripts/get_latest_build_number.sh)
+  BUILD_NUMBER=$(($LATEST_BUILD_NUMBER + 1))
+  log "Build number to use for the beta build: $BUILD_NUMBER"
+fi
+
+log "Generating env file required by Fastlane..."
+tee .env <<EOF
+export BRANCH_TO_BUILD=${BRANCH_TO_BUILD}
+export GIT_LOCAL_BRANCH=${GIT_LOCAL_BRANCH}
+export COMMIT_CHANGES_TO_GIT=true
+$([ -z "$BUMP_BUILD_NUMBER" ] || echo "\
+export INCREMENT_BUILD_NUMBER=true
+export BUILD_NUMBER=${BUILD_NUMBER}
+")
+$([ -z "$BUMP_VERSION_NUMBER" ] || echo "\
+export INCREMENT_VERSION_NUMBER=true
+export VERSION=${VERSION}
+")
+EOF
+
+log "Setting up fastlane environment"
+(. .env && cd fastlane && bundle install)
```

Comment:
I am a bit rusty with shell scripts, so just double checking.

If any of these commands fail, the whole process will fail, right? It is not going to happen anything like "bundle install fails, I will try to set the app build number eitherway".

Reply:
I double checked this, and inserting a failing command at any point (e.g. `(. .env && false && cd fastlane && ...)`) will cause the whole script to fail, since the subshell exits with code 1, and the parent script has the `set -e` option enabled